src/globus_sdk/services/flows/errors.py

```diff
@@ -1,7 +1,39 @@
-from globus_sdk.exc import GlobusAPIError
+from __future__ import annotations
+
+from globus_sdk.exc import ErrorSubdocument, GlobusAPIError
 
 
 class FlowsAPIError(GlobusAPIError):
     """
     Error class to represent error responses from Flows.
     """
+
+    def _parse_undefined_error_format(self) -> bool:
```

Comment:
Question, not suggestion:

Would it make sense to explicitly add a "custom parse" overridable method like

```
class GlobusAPIError:
   ...
   def _custom_parse(self) -> bool:
       return False
```

So order of parsing would be like:

* JsonAPI
* TypeZero
* Custom (if exists)
* undefined

Reply:
Having written that, I'm still thinking about this.

Let me see how messy and hard it would be to rig this up as `() -> bool | None`.
Part of the need for `None` as a supported output is to indicate that parsing didn't detect whatever custom thing was wanted, and that the default undefined parsing needs to take over.

i.e. Calls to `super()._parse_undefined_error_format()` could become `return None`.

I'll give it a quick try, but I might ditch it if I don't like it or it turns out to be too hard.

Reply:
If custom parse fails wouldn't we want to fall back to the default undefined parsing handler?

Reply:
I ditched it for now. I'm happy to circle back later though.

The end result of my first crack at this was that `return None` became a slightly obscure alias for `return super()._parse_undefined_error_format()`.

---

This is the sort of comparison I ended up with:
```python
if not isinstance(self._dict_data.get("error"), dict):
    return None
...
```
vs
```python
if not isinstance(self._dict_data.get("error"), dict):
    return super()._parse_undefined_error_format()
...
```

The path I was taking was that parsing needs to be able to signal success, failure, or "I don't know, call my parent's way of handling this", and that idea is better captured by overriding the undefined handler.